src/Geometry.cpp

```diff
@@ -528,48 +519,49 @@ int Geometry::findFSRId(LocalCoords* coords) {
   std::size_t fsr_key_hash = key_hash_function(getFSRKey(coords));
 
   /* If FSR has not been encountered, update FSR maps and vectors */
-  if (_FSR_keys_map.find(fsr_key_hash) == _FSR_keys_map.end()){
-
-    /* Get the cell that contains coords */
-    Cell* cell = findCellContainingCoords(curr);
-    
-    /* Get the lock */
-    omp_set_lock(_num_FSRs_lock);
-
-    /* Recheck to see if FSR has been added to maps after getting the lock */
-    if (_FSR_keys_map.find(fsr_key_hash) != _FSR_keys_map.end())
-      fsr_id = _FSR_keys_map.at(fsr_key_hash)._fsr_id;
+  if (!_FSR_keys_map.contains(fsr_key_hash)){
+
+    /* Try to get a clean copy of the fsr_id, adding the FSR data 
+       if necessary where -1 indicates the key was already added */
+    fsr_id = _FSR_keys_map.insert_and_get_count(fsr_key_hash, NULL);
+    if( fsr_id == -1)
+    {
+      fsr_data volatile* fsr;
+      do{
```

Comment:
Yes, there are inconsistencies all over OpenMOC with regard to the space before ``{``.

If I type
```
grep -r \){ src
```
from the ``OpenMOC`` directory, it returns:
```
src/accel/cuda/GPUSolver.cu:  for (int e=0; e < _num_groups; e++){
src/accel/cuda/GPUSolver.cu:  for (int g=0; g < _num_groups; g++){
src/accel/cuda/GPUSolver.cu:void GPUSolver::initializeExpEvaluator(){
src/accel/cuda/GPUSolver.cu:    for (iter=host_materials.begin(); iter != host_materials.end(); ++iter){
src/log.cpp:int get_log_level(){
src/linalg.h:inline void vector_copy(T* vector_from, T* vector_to, int length){
src/linalg.h:inline void matrix_zero(T** matrix, int width, int length){
src/linalg.h:  for (int i = 0; i < length; i++){
src/linalg.h:inline void vector_zero(T* vector, int length){
src/linalg.h:                                  int block_width){
src/linalg.h:  for (int i = 0; i < num_blocks; i++){
src/linalg.h:    for (int g = 0; g < block_width; g++){
src/linalg.h:      for (int e = 0; e < block_width; e++){
src/linalg.h:inline void vector_scale(T* vector, T scale_value, int length){
src/ExpEvaluator.cpp:  for (int i=0; i < num_array_values; i++){
src/ExpEvaluator.cpp:    for (int p=0; p < num_polar; p++){
src/Geometry.cpp.orig:Cmfd* Geometry::getCmfd(){
src/Geometry.cpp.orig:void Geometry::setCmfd(Cmfd* cmfd){
src/Geometry.cpp.orig:  if (univ->getId() == _root_universe->getId()){
src/Geometry.cpp.orig:    if (_cmfd != NULL){
src/Geometry.cpp.orig:  if (!_FSR_keys_map.contains(fsr_key_hash)){
src/Geometry.cpp.orig:      if (_cmfd != NULL){
src/Geometry.cpp.orig:  if (_cmfd != NULL){
src/Geometry.cpp.orig:  while(curr != NULL){
src/Geometry.cpp.orig:    if (_cmfd != NULL){
src/Geometry.cpp.orig:void Geometry::initializeFSRVectors(){
src/Geometry.cpp.orig:void Geometry::initializeCmfd(){
src/Geometry.cpp.orig:ParallelHashMap<std::size_t, fsr_data*>* Geometry::getFSRKeysMap(){
src/Geometry.cpp.orig:std::unordered_map<std::size_t, fsr_data>* Geometry::getFSRKeysMap(){
src/Geometry.cpp.orig:std::vector<std::size_t>* Geometry::getFSRsToKeys(){
src/Geometry.cpp.orig:                             FSR_keys_map){
src/Geometry.cpp.orig:void Geometry::setFSRsToKeys(std::vector<std::size_t>* FSRs_to_keys){
src/Geometry.cpp.orig:void Geometry::setFSRsToMaterialIDs(std::vector<int>* FSRs_to_material_IDs){
src/Geometry.cpp.orig:bool Geometry::withinBounds(LocalCoords* coords){
src/Geometry.cpp.orig:void Geometry::setFSRCentroid(int fsr, Point* centroid){
src/Geometry.cpp.orig:Cell* Geometry::findCellContainingFSR(int fsr_id){
src/Surface.cpp:Surface::Surface(const int id, const char* name){
src/Surface.cpp:boundaryType Surface::getBoundaryType(){
src/Surface.cpp:double Plane::getMinX(int halfspace){
src/Surface.cpp:double Plane::getMaxX(int halfspace){
src/Surface.cpp:double Plane::getMinY(int halfspace){
src/Surface.cpp:double Plane::getMaxY(int halfspace){
src/Surface.cpp:double Plane::getMinZ(int halfspace){
src/Surface.cpp:double Plane::getMaxZ(int halfspace){
src/Surface.cpp:double XPlane::getMinX(int halfspace){
src/Surface.cpp:double XPlane::getMaxX(int halfspace){
src/Surface.cpp:double YPlane::getMinY(int halfspace){
src/Surface.cpp:double YPlane::getMaxY(int halfspace){
src/Surface.cpp:double ZPlane::getMinZ(int halfspace){
src/Surface.cpp:double ZPlane::getMaxZ(int halfspace){
src/Surface.cpp:double Circle::getMinX(int halfspace){
src/Surface.cpp:double Circle::getMaxX(int halfspace){
src/Surface.cpp:double Circle::getMinY(int halfspace){
src/Surface.cpp:double Circle::getMaxY(int halfspace){
src/Surface.cpp:double Circle::getMinZ(int halfspace){
src/Surface.cpp:double Circle::getMaxZ(int halfspace){
src/ParallelHashMap.h:    node(K k_in, V v_in) : next(NULL), key(k_in), value(v_in){}
src/CPUSolver.cpp:                                FP_PRECISION* fsr_flux){
src/CPUSolver.cpp:    for (int p=0; p < _num_polar; p++){
src/CPUSolver.cpp:                                    FP_PRECISION* track_flux, bool fwd){
src/LocalCoords.cpp:    else if (curr->getType() == LAT){
src/Solver.cpp:void Solver::initializeCmfd(){
src/Solver.cpp:    if (_cmfd != NULL && _cmfd->isFluxUpdateOn()){
src/Cmfd.cpp:  if (_M != NULL){
src/Cmfd.cpp:  if (_A != NULL){
src/Cmfd.cpp:void Cmfd::setNumX(int num_x){
src/Cmfd.cpp:void Cmfd::setNumY(int num_y){
src/Cmfd.cpp:int Cmfd::getNumX(){
src/Cmfd.cpp:int Cmfd::getNumY(){
src/Cmfd.cpp:void Cmfd::setWidth(double width){
src/Cmfd.cpp:void Cmfd::setHeight(double height){
src/Cmfd.cpp:void Cmfd::computeXS(){
src/Cmfd.cpp:  for (int i = 0; i < _num_x * _num_y; i++){
src/Cmfd.cpp:      for (int g = 0; g < _num_cmfd_groups; g++){
src/Cmfd.cpp:           iter != _cell_fsrs.at(i).end(); ++iter){
src/Cmfd.cpp:        for (int b = 0; b < _num_cmfd_groups; b++){
src/Cmfd.cpp:          for (int h = 0; h < _num_moc_groups; h++){
src/Cmfd.cpp:      for (int h = _group_indices[e]; h < _group_indices[e+1]; h++){
src/Cmfd.cpp:             iter != _cell_fsrs.at(i).end(); ++iter){
src/Cmfd.cpp:          for (int g = 0; g < _num_moc_groups; g++){
src/Cmfd.cpp:      for (int g = 0; g < _num_cmfd_groups; g++){
src/Cmfd.cpp:void Cmfd::computeDs(int moc_iteration){
src/Cmfd.cpp:  for (int y = 0; y < _num_y; y++){
src/Cmfd.cpp:    for (int x = 0; x < _num_x; x++){
src/Cmfd.cpp:      for (int surface = 0; surface < 4; surface++){
src/Cmfd.cpp:        for (int e = 0; e < _num_cmfd_groups; e++){
src/Cmfd.cpp:          if (surface == 0 || surface== 2){
src/Cmfd.cpp:          else if (surface == 1 || surface == 3){
src/Cmfd.cpp:          if (cell_next == -1){
src/Cmfd.cpp:            if (_boundaries[surface] == REFLECTIVE){
src/Cmfd.cpp:            else if (_boundaries[surface] == VACUUM){
src/Cmfd.cpp:            if (surface == 0){
src/Cmfd.cpp:            else if (surface == 1){
src/Cmfd.cpp:            else if (surface == 2){
src/Cmfd.cpp:            else if (surface == 3){
src/Cmfd.cpp:            if (fabs(d_tilde) > fabs(d_hat) && moc_iteration != 0){
src/Cmfd.cpp:              if (sense == -1){
src/Cmfd.cpp:                if (1 - fabs(d_tilde)/d_tilde < 1e-8){
src/Cmfd.cpp:                if (1 - fabs(d_tilde)/d_tilde < 1e-8){
src/Cmfd.cpp:FP_PRECISION Cmfd::computeKeff(int moc_iteration){
src/Cmfd.cpp:  if (_A == NULL){
src/Cmfd.cpp:      for (int i = 0; i < _num_x*_num_y; i++){
src/Cmfd.cpp:    catch(std::exception &e){
src/Cmfd.cpp:  for (int iter = 0; iter < 25000; iter++){
src/Cmfd.cpp:    for (int i = 0; i < _num_x*_num_y*_num_cmfd_groups; i++){
src/Cmfd.cpp:                       FP_PRECISION* vec_b, FP_PRECISION conv, int max_iter){
src/Cmfd.cpp:  while (iter < max_iter){
src/Cmfd.cpp:    for (int y = 0; y < _num_y; y++){
src/Cmfd.cpp:      for (int x = y % 2; x < _num_x; x += 2){
src/Cmfd.cpp:        for (int g = 0; g < _num_cmfd_groups; g++){
src/Cmfd.cpp:          for (int e = 0; e < _num_cmfd_groups; e++){
src/Cmfd.cpp:    for (int y = 0; y < _num_y; y++){
src/Cmfd.cpp:      for (int x = 1 - y % 2; x < _num_x; x += 2){
src/Cmfd.cpp:        for (int g = 0; g < _num_cmfd_groups; g++){
src/Cmfd.cpp:          for (int e = 0; e < _num_cmfd_groups; e++){
src/Cmfd.cpp:    for (int i = 0; i < _num_x*_num_y*_num_cmfd_groups; i++){
src/Cmfd.cpp:void Cmfd::rescaleFlux(){
src/Cmfd.cpp:void Cmfd::constructMatrices(){
src/Cmfd.cpp:  for (int y = 0; y < _num_y; y++){
src/Cmfd.cpp:    for (int x = 0; x < _num_x; x++){
src/Cmfd.cpp:      for (int e = 0; e < _num_cmfd_groups; e++){
src/Cmfd.cpp:        for (int g = 0; g < _num_cmfd_groups; g++){
src/Cmfd.cpp:          if (e != g){
src/Cmfd.cpp:        if (x != _num_x - 1){
src/Cmfd.cpp:        if (x != 0){
src/Cmfd.cpp:        if (y != 0){
src/Cmfd.cpp:        if (y != _num_y - 1){
src/Cmfd.cpp:        for (int g = 0; g < _num_cmfd_groups; g++){
src/Cmfd.cpp:void Cmfd::updateMOCFlux(){
src/Cmfd.cpp:  for (int i = 0; i < _num_x*_num_y; i++){
src/Cmfd.cpp:    for (int e = 0; e < _num_cmfd_groups; e++){
src/Cmfd.cpp:      for (int h = _group_indices[e]; h < _group_indices[e+1]; h++){
src/Cmfd.cpp:FP_PRECISION Cmfd::computeDiffCorrect(FP_PRECISION d, FP_PRECISION h){
src/Cmfd.cpp:  if (_optically_thick){
src/Cmfd.cpp:    for (int p = 0; p < _num_polar; p++){
src/Cmfd.cpp:void Cmfd::setFSRMaterials(Material** FSR_materials){
src/Cmfd.cpp:void Cmfd::setFSRVolumes(FP_PRECISION* FSR_volumes){
src/Cmfd.cpp:void Cmfd::setFSRFluxes(FP_PRECISION* scalar_flux){
src/Cmfd.cpp:void Cmfd::setSORRelaxationFactor(FP_PRECISION SOR_factor){
src/Cmfd.cpp:int Cmfd::getNumCmfdGroups(){
src/Cmfd.cpp:int Cmfd::getCmfdGroup(int group){
src/Cmfd.cpp:void Cmfd::setGroupStructure(int* group_indices, int length_group_indices){
src/Cmfd.cpp:  if (_group_indices == NULL){
src/Cmfd.cpp:  if (group_indices == NULL){
src/Cmfd.cpp:    for (int i = 0; i < length_group_indices; i++){
src/Cmfd.cpp:    for (int i = 1; i < length_group_indices; i++){
src/Cmfd.cpp:void Cmfd::initializeFlux(){
src/Cmfd.cpp:  catch(std::exception &e){
src/Cmfd.cpp:  for (int y = 0; y < _num_y; y++){
src/Cmfd.cpp:    for (int x = 0; x < _num_x; x++){
src/Cmfd.cpp:      for (int g = 0; g < _num_cmfd_groups; g++){
src/Cmfd.cpp:void Cmfd::initializeMaterials(){
src/Cmfd.cpp:    for (int y = 0; y < _num_y; y++){
src/Cmfd.cpp:      for (int x = 0; x < _num_x; x++){
src/Cmfd.cpp:  catch(std::exception &e){
src/Cmfd.cpp:void Cmfd::initializeCellMap(){
src/Cmfd.cpp:  for (int y = 0; y < _num_y; y++){
src/Cmfd.cpp:    for (int x = 0; x < _num_x; x++){
src/Cmfd.cpp:void Cmfd::initializeGroupMap(){
src/Cmfd.cpp:  if (_group_indices_map == NULL){
src/Cmfd.cpp:  for (int e = 0; e < _num_cmfd_groups; e++){
src/Cmfd.cpp:    for (int h = _group_indices[e]; h < _group_indices[e+1]; h++){
src/Cmfd.cpp:int Cmfd::findCmfdSurface(int cell, LocalCoords* coords){
src/Cmfd.cpp:int Cmfd::findCmfdCell(LocalCoords* coords){
src/Cmfd.cpp:void Cmfd::setLattice(Lattice* lattice){
src/Cmfd.cpp:void Cmfd::setLatticeStructure(int num_x, int num_y){
src/Cmfd.cpp:Lattice* Cmfd::getLattice(){
src/Cmfd.cpp:void Cmfd::addFSRToCell(int cmfd_cell, int fsr_id){
src/Cmfd.cpp:void Cmfd::setNumMOCGroups(int num_groups){
src/Cmfd.cpp:int Cmfd::getNumMOCGroups(){
src/Cmfd.cpp:int Cmfd::getNumCells(){
src/Cmfd.cpp:void Cmfd::setNumFSRs(int num_fsrs){
src/Cmfd.cpp:void Cmfd::splitCorners(){
src/Cmfd.cpp:  for (int x = 0; x < _num_x; x++){
src/Cmfd.cpp:    for (int y = 0; y < _num_y; y++){
src/Cmfd.cpp:      if (x > 0 && y > 0){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      else if (x == 0 && y != 0){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      else if (x != 0 && y == 0){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      if (x < _num_x - 1 && y > 0){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      else if (x == _num_x - 1 && y > 0){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      else if (x < _num_x - 1 && y == 0){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      if (x < _num_x - 1 && y < _num_y - 1){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      else if (x == _num_x - 1 && y != _num_y - 1){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      else if (x != _num_x - 1 && y == _num_y - 1){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      if (x > 0 && y < _num_y - 1){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      else if (x == 0 && y != _num_y - 1){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      else if (x != 0 && y == _num_y - 1){
src/Cmfd.cpp:        for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:      for (int e = 0; e < ncg; e++){
src/Cmfd.cpp:int Cmfd::getCellNext(int cell_num, int surface_id){
src/Cmfd.cpp:  if (surface_id == 0){
src/Cmfd.cpp:  else if (surface_id == 1){
src/Cmfd.cpp:  else if (surface_id == 2){
src/Cmfd.cpp:  else if (surface_id == 3){
src/Cmfd.cpp:bool Cmfd::isOpticallyThick(){
src/Cmfd.cpp:void Cmfd::setOpticallyThick(bool optically_thick){
src/Cmfd.cpp:FP_PRECISION Cmfd::getMOCRelaxationFactor(){
src/Cmfd.cpp:void Cmfd::setMOCRelaxationFactor(FP_PRECISION relax_factor){
src/Cmfd.cpp:void Cmfd::setBoundary(int side, boundaryType boundary){
src/Cmfd.cpp:int Cmfd::getBoundary(int side){
src/Cmfd.cpp:int Cmfd::convertFSRIdToCmfdCell(int fsr_id){
src/Cmfd.cpp:  for (int cell=0; cell < _num_x*_num_y; cell++){
src/Cmfd.cpp:           iter != _cell_fsrs.at(cell).end(); ++iter){
src/Cmfd.cpp:std::vector< std::vector<int> > Cmfd::getCellFSRs(){
src/Cmfd.cpp:void Cmfd::setCellFSRs(std::vector< std::vector<int> > cell_fsrs){
src/Cmfd.cpp:void Cmfd::setFluxUpdateOn(bool flux_update_on){
src/Cmfd.cpp:bool Cmfd::isFluxUpdateOn(){
src/Cmfd.cpp:FP_PRECISION Cmfd::getFluxRatio(int cmfd_cell, int moc_group){
src/Cmfd.cpp:			      int num_tracks){
src/Cmfd.cpp:    if (bc){
src/Cmfd.cpp:    if (bc){
src/Cmfd.cpp:  if (curr_segment->_cmfd_surface_fwd != -1 && fwd){
src/Cmfd.cpp:  else if (curr_segment->_cmfd_surface_bwd != -1 && !fwd){
src/VectorizedSolver.cpp:                                       FP_PRECISION* fsr_flux){
src/VectorizedSolver.cpp:  for (int p=0; p < _num_polar; p++){
src/TrackGenerator.cpp:  if (_geometry->getCmfd() != NULL){
src/TrackGenerator.cpp:      for (int j=0; j < _num_tracks[i]; j++){
src/TrackGenerator.cpp:        if (cmfd != NULL){
src/TrackGenerator.cpp:  for (int i=0; i < num_FSRs; i++){
src/TrackGenerator.cpp:  if (cmfd != NULL){
src/TrackGenerator.cpp:    for (int cell=0; cell < num_cells; cell++){
src/TrackGenerator.cpp:        if (cmfd != NULL){
src/TrackGenerator.cpp:  for (int fsr_id=0; fsr_id < num_FSRs; fsr_id++){
src/TrackGenerator.cpp:  if (cmfd != NULL){
src/TrackGenerator.cpp:    for (int cell=0; cell < num_cells; cell++){
src/TrackGenerator.cpp:      for (int fsr = 0; fsr < num_FSRs; fsr++){
src/TrackGenerator.cpp:void TrackGenerator::generateFSRCentroids(){
src/TrackGenerator.cpp:  for (int r=0; r < num_FSRs; r++){
src/Universe.cpp:double Universe::getMaxY(){
src/Universe.cpp:double Lattice::getMaxY(){
src/Universe.cpp:  for (iter = unique_universes.begin(); iter != unique_universes.end(); ++iter){
src/Universe.cpp:  for (iter = unique_universes.begin(); iter != unique_universes.end(); ++iter){
src/Universe.cpp:    for (int i = 0; i < _num_x; i++){
src/Universe.cpp:int Lattice::getLatticeCell(Point* point){
src/Universe.cpp:  if (fabs(x - left) <= ON_SURFACE_THRESH){
src/Universe.cpp:  else if (fabs(x - right) <= ON_SURFACE_THRESH){
src/TrackGenerator.cpp.orig:  if (_geometry->getCmfd() != NULL){
src/TrackGenerator.cpp.orig:      for (int j=0; j < _num_tracks[i]; j++){
src/TrackGenerator.cpp.orig:        if (cmfd != NULL){
src/TrackGenerator.cpp.orig:  for(int i=0; i < num_FSRs; i++){
src/TrackGenerator.cpp.orig:  for (iter = FSR_keys_map->begin(); iter != FSR_keys_map->end(); ++iter){
src/TrackGenerator.cpp.orig:  if (cmfd != NULL){
src/TrackGenerator.cpp.orig:    for (int cell=0; cell < num_cells; cell++){
src/TrackGenerator.cpp.orig:        if (cmfd != NULL){
src/TrackGenerator.cpp.orig:  for (int fsr_id=0; fsr_id < num_FSRs; fsr_id++){
src/TrackGenerator.cpp.orig:  if (cmfd != NULL){
src/TrackGenerator.cpp.orig:    for (int cell=0; cell < num_cells; cell++){
src/TrackGenerator.cpp.orig:      for (int fsr = 0; fsr < num_FSRs; fsr++){
src/TrackGenerator.cpp.orig:void TrackGenerator::generateFSRCentroids(){
src/TrackGenerator.cpp.orig:  for (int r=0; r < num_FSRs; r++){
src/Material.cpp:  if (_dif_coef == NULL){
src/Material.cpp:  if (_dif_hat == NULL){
src/Material.cpp:  if (_dif_tilde == NULL){
src/Material.cpp:  if (_buckling == NULL){
src/Material.cpp:  for (int i=0; i < _num_groups; i++){
src/Material.cpp:  for (int i=0; i < _num_groups; i++){
src/Material.cpp:  if (_dif_coef == NULL){
src/Material.cpp:  if (_buckling == NULL){
src/Material.cpp:  if (_dif_hat == NULL){
src/Material.cpp:  if (_dif_tilde == NULL){
src/Material.cpp:  if (_dif_tilde == NULL){
src/Material.cpp:Material* Material::clone(){
```